src/emeus-constraint-layout.c

```diff
@@ -1138,6 +1155,34 @@ remove_child_constraint (EmeusConstraintLayout      *layout,
   return TRUE;
 }
 
+void
+emeus_constraint_layout_activate_constraint (EmeusConstraintLayout *layout,
+                                             EmeusConstraint       *constraint)
+{
+  g_assert (constraint->solver == &layout->solver);
+
+  if (constraint->constraint == NULL)
+    return;
+
+  if (constraint->target_object == layout)
+    create_layout_constraint (layout, constraint);
+  else
+    create_child_constraint (layout, constraint->target_object, constraint);
+}
+
+void
+emeus_constraint_layout_deactivate_constraint (EmeusConstraintLayout *layout,
+                                               EmeusConstraint       *constraint)
+{
+  g_assert (constraint->solver == &layout->solver);
+
+  if (constraint->constraint == NULL)
+    return;
+
+  simplex_solver_remove_constraint (constraint->solver, constraint->constraint);
+  constraint->constraint = NULL;
```

Comment:
Oh, nicely spotted: I inverted the condition. The `activate_constraint()` function should return if there's a constraint already set, because it means that the constraint is already active.